pkg/ipcache/ipcache.go

```diff
@@ -529,7 +530,12 @@ func (ipc *IPCache) UpsertMetadataBatch(updates ...MU) (revision uint64) {
 	prefixes := make([]cmtypes.PrefixCluster, 0, len(updates))
 	ipc.metadata.Lock()
 	for _, upd := range updates {
-		prefixes = append(prefixes, ipc.metadata.upsertLocked(upd.Prefix, upd.Source, upd.Resource, upd.Metadata...)...)
+		switch {
+		case !upd.IsCIDR:
+			fallthrough
```

Comment:
This switch is somewhat hard to read :sweat_smile: AFAICT it's equivalent to the following

```go
// Only upsert CIDR if we haven't seen it before.
if !udp.IsCIDR || ipc.metadata.prefixRefCounter.Add(udp.Prefix) {
    prefixes = append(prefixes, ipc.metadata.upsertLocked(upd.Prefix, upd.Source, upd.Resource, upd.Metadata...)...)
}
```